package.json

```diff
@@ -30,27 +30,36 @@
   "homepage": "http://fluidproject.org",
   "dependencies": {
     "es6-template-strings": "2.0.0",
-    "extend": "^3.0.0"
+    "extend": "3.0.0"
   },
   "devDependencies": {
-    "grunt": "^0.4.5",
-    "grunt-contrib-jshint": "~0.9.0",
+    "grunt": "1.0.1",
+    "eslint-config-fluid": "1.0.0",
+    "fluid-grunt-eslint": "18.1.2",
     "grunt-jsonlint": "1.0.4",
-    "sinon": "^1.17.1"
+    "sinon": "1.17.1"
   },
   "defaultOptions": {
     "changesCmd": "git status -s -uno",
+    "checkRemoteCmd": "git ls-remote --exit-code ${remote}",
     "rawTimestampCmd": "git show -s --format=%ct HEAD",
     "revisionCmd": "git rev-parse --verify --short HEAD",
     "packCmd": "npm pack",
     "publishCmd": "npm publish",
+    "publishDevCmd": "npm publish --tag ${devTag}",
     "versionCmd": "npm version --no-git-tag-version ${version}",
-    "distTagCmd": "npm dist-tag add ${packageName}@${version} ${tag}",
     "cleanCmd": "git checkout -- package.json",
     "vcTagCmd": "git tag -a v${version} -m 'Tagging the ${version} release'",
-    "pushVCTagCmd": "git push upstream v${version}",
+    "pushVCTagCmd": "git push ${remote} v${version}",
     "devVersion": "${version}-${preRelease}.${timestamp}.${revision}",
     "devTag": "dev",
-    "moduleRoot": ""
+    "remoteName": "upstream",
+    "moduleRoot": "",
+    "changesHint": "Address uncommitted changes: Commit \"git commit -a\", Stash \"git stash\" or Clean \"git reset --hard\"\n",
+    "checkRemoteHint": "Run \"git remote -v\" for a list of available remote repositories.\n",
+    "publishHint": "Ensure that you have access to publish to the registry and that the current version does not already exist.\n",
+    "publishDevHint": "Ensure that you have access to publish to the registry and that the current version does not already exist.\nIf the tag already exists use a new tag name or run \"npm dist-tag rm ${packageName} ${tag}\" to remove the existing one.\nTags share a namespace with versions. To reduce confusion, tags that can be interpreted as semver values, will be rejected.\n",
```

Comment:
@amb26 yes, this refers to NPM tags, not version control tags. The command used is `publishDevCmd": "npm publish --tag ${devTag}` and devTag defaults to "dev". However, if a user were to override the dev tag to say "v2.2" or something of that sort, it would be rejected by NPM. So the rejection is not happening by fluid-publish but by NPM.